components/butterfly/src/server/mod.rs

```diff
@@ -64,6 +65,31 @@ use trace::{Trace, TraceKind};
 /// down and leave the ring.
 const SELF_DEPARTURE_RUMOR_FANOUT: usize = 10;
 
+lazy_static! {
+    static ref INCARNATION: Gauge = register_gauge!(opts!(
```

Comment:
From what I can tell after reading through the docs and code a bit more carefully, while this is definitely possible, making it happens means deviating from the `register_x!` macros (or writing our own) and setting everything up manually.  That's certainly something we could do, but it doesn't feel worth it to me in terms of both effort and consistency in our code, just to avoid a conversion from `u64` -> `i64`.  Maybe that's worth exploring in the future though.

Reply:
Yeah, I tend to agree. The savings would mostly be in the elimination of the conversion code added to core, but if that comes at the cost of more setup code, it seems like a wash. A PR to the `prometheus` crate to make it easy to use `u64`-based types could be nice, but that's outside the scope of this review.